### PR TITLE
Fix problem when odo deploy processes do not terminate

### LINKED ISSUE #6715: Integration test suite taking 30 minutes to terminate
/kind bug

On Kubernetes integration tests, I can see the job is blocked for ~30 minutes after finishing the first series of tests, at this stage:
```
Ran 436 of 788 Specs in 1204.650 seconds
SUCCESS! -- 436 Passed | 0 Failed | 0 Pending | 352 Skipped

Ginkgo ran 1 suite in 50m40.411198089s
```

We can see the difference between the 1204 seconds (20 minutes) to execute all the tests and the 50 minutes to finish the suite: 30 minutes lost

This does not happen on OpenShift, and we can see the difference of execution time (50 minutes on Kubernetes vs 22 minutes on OpenShift):

```
Ran 436 of 793 Specs in 1301.953 seconds
SUCCESS! -- 436 Passed | 0 Failed | 0 Pending | 357 Skipped

Ginkgo ran 1 suite in 22m47.524234241s
```

Here, the difference between the 1301 seconds and 22 minutes is only 1 minute.

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind bug
/area testing
**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6715 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: tests/helper/helper_cmd_wrapper.go [hunk 1]
```diff
@@ -117,6 +117,7 @@ func (cw *CmdWrapper) ShouldRun() *CmdWrapper {
 func (cw *CmdWrapper) Should(f func(session *gexec.Session)) {
 	cw.Runner()
 	f(cw.session)
+	cw.session.Wait()
```
Comment (valaparthvi) [inline tests/helper/helper_cmd_wrapper.go:120]: This seems to fix the problem, logs point so.
```sh
Ran 436 of 788 Specs in 1079.759 seconds
SUCCESS! -- 436 Passed | 0 Failed | 0 Pending | 352 Skipped


Ginkgo ran 1 suite in 19m5.388476487s
Test Suite Passed
```
But I don't understand why this missing piece created a problem in the first place.


I also noticed that if you check this run on IBM cloud, the duration for Job run was `Duration1 hour, 57 minutes, 19 seconds`, but the test run duration was `Duration23 minutes, 40 seconds`, is that normal?

Comment (feloy) [reply tests/helper/helper_cmd_wrapper.go:120]: > But I don't understand why this missing piece created a problem in the first place.

The tests were terminating before the processes `odo deploy` were terminated. Ginkgo probably has problems handling children processes remaining after the tests finish.

> I also noticed that if you check this run on IBM cloud, the duration for Job run was Duration1 hour, 57 minutes, 19 seconds, but the test run duration was Duration23 minutes, 40 seconds, is that normal?

The job waited 1h34 before to start.

**Pull request merged**